bench/bench_lib.py

```
@@ -194,16 +206,77 @@ def compare_summaries(baseline, candidate):
+    """Paired per-query comparison: per metric, each query's median in each run.
+
+    The bench runs a fixed query list, so every candidate row has a matching
+    baseline row. Comparing them pairwise is the only reading of this bench
+    that survives both of its distortions.
+
+    A pooled median across all rows sits on a cluster boundary — the queries
+    span ~31k to ~69k tokens — and flips on noise: the 2026-07-29 pair moved
+    +14.2% while every per-query median was flat within ±0.7%. A pooled sum is
+    just as fragile in the other direction: one runaway run (q06 repeat 3,
+    180,035 tokens against a ~46,700 typical) moved the total by more than the
+    entire real difference, printing −8.2% on two runs that were the same.
```

Ruff (0.16.0)

[warning] Docstring contains ambiguous `−` (MINUS SIGN). Did you mean `-` (HYPHEN-MINUS)?

(RUF002)